dev/devicelab/lib/tasks/plugin_tests.dart

```diff
@@ -146,14 +150,37 @@ class _FlutterProject {
     await inDirectory(Directory(rootPath), () async {
       final String buildOutput =  await evalFlutter('build', options: <String>[target, '-v']);
 
-      if (target == 'ios') {
+      if (target == 'ios' || target == 'macos') {
         // This warning is confusing and shouldn't be emitted. Plugins often support lower versions than the
         // Flutter app, but as long as they support the minimum it will work.
         // warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0,
         // but the range of supported deployment target versions is 9.0 to 14.0.99.
+        //
+        // (or "The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET'"...)
         if (buildOutput.contains('the range of supported deployment target versions')) {
           throw TaskResult.failure('Minimum plugin version warning present');
         }
+
+        final File podsProject = File(path.join(rootPath, target, 'Pods', 'Pods.xcodeproj', 'project.pbxproj'));
+        if (!podsProject.existsSync()) {
+          throw TaskResult.failure('Xcode Pods project file missing at ${podsProject.path}');
+        }
+
+        final String podsProjectContent = podsProject.readAsStringSync();
+        // This may be a bit brittle, IPHONEOS_DEPLOYMENT_TARGET appears in the
+        // Pods Xcode project file 6 times. If this number changes, make sure
+        // it's not a regression in the IPHONEOS_DEPLOYMENT_TARGET override logic.
+        // The plugintest target should not have IPHONEOS_DEPLOYMENT_TARGET set.
+        // See _reduceDarwinPluginMinimumVersion for details.
+        if (target == 'ios' && 'IPHONEOS_DEPLOYMENT_TARGET'.allMatches(podsProjectContent).length != 6) {
```

Comment:
@stuartmorgan I started out actually parsing the Xcode project for the `plugintest` target's build settings but that got pretty confusing.  This count check is a bit of a cheat but it would definitely fail if the behavior was wrong or regressed...

Reply:
Makes sense; I was thinking it wouldn't be in the project at all, so I didn't realize it would be non-trivial. Agreed that a full parser would be overkill here.